### PR TITLE
net-mgmt/zabbix-agent: add Zabbix Agent to plugins

### PR DESCRIPTION
**Pull request by fraenki**
From the authors of Zabbix:

> Zabbix is the ultimate enterprise-level software designed for real-time monitoring of millions of metrics collected from tens of thousands of servers, virtual machines and network devices.
> Zabbix is Open Source and comes at no cost.

This will add the Zabbix Agent to our growing list of plugins, effectively allowing OPNsense to be monitored with Zabbix.

This version of the plugin is ready for use, but not feature complete (from my point of view). Thus I've set the version to 0.1 for now.

The wishlist for version 1.0 (or later):
- allow to create UserParameter entries (arbitrary shell commands, run as Zabbix user)
- provide a set of pre-defined OPNsense checks (pf stats, IPsec status, configd commands, OPNsense API calls)
- automatically generate firewall rules from ListenIP/ListenPort and Servers/ActiveServers
- fix the log file in GUI (see #69)

### CODE REVIEW

File: net-mgmt/zabbix-agent/src/etc/inc/plugins.inc.d/zabbixagent.inc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+/**
+XXX: needs investigation, hostname must be unique/replaced/excluded on secondary node(s)
+function zabbixagent_xmlrpc_sync()
+{
+    $result = array();
+    $result['id'] = 'zabbixagent';
+    $result['section'] = 'OPNsense.zabbixagent';
+    $result['description'] = gettext('Enterprise-class open source distributed monitoring agent');
+    return array($result);
+}
```
Comment (fabianfrz) [inline net-mgmt/zabbix-agent/src/etc/inc/plugins.inc.d/zabbixagent.inc]: Seems to be not final.

File: net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/zabbix_agentd.conf [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+# TLSPSKIdentity=
+
+### Option: TLSPSKFile
+#	Full pathname of a file containing the pre-shared key.
+#
+# Mandatory: no
+# Default:
+# TLSPSKFile=
+
+{% endif %}
```
Comment (fabianfrz) [inline net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/zabbix_agentd.conf]: In a generated configuration file, documentation comments are bloat.

File: net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/zabbix_agentd.conf [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+#	If port is not specified, default port is used.
+#	IPv6 addresses must be enclosed in square brackets if port for that host is specified.
+#	If port is not specified, square brackets for IPv6 addresses are optional.
+#	If this parameter is not specified, active checks are disabled.
+#	Example: ServerActive=127.0.0.1:20051,zabbix.domain,[::1]:30051,::1,[12fc::1]
+#
+{% if OPNsense.ZabbixAgent.settings.features.enableActiveChecks == '1' and OPNsense.ZabbixAgent.settings.features.activeCheckServers|default("") != "" %}
+ServerActive={{OPNsense.ZabbixAgent.settings.features.activeCheckServers}}
+{% else %}
+# NOTE: Active checks are disabled
```
Comment (fabianfrz) [inline net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/zabbix_agentd.conf]: useless control path

File: net-mgmt/zabbix-agent/src/www/diag_logs_zabbixagent.php [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+$logfile = '/var/log/zabbix/zabbix_agentd.log';
+$logclog = false;
+
+require_once 'diag_logs_template.inc';
```
Comment (fabianfrz) [inline net-mgmt/zabbix-agent/src/www/diag_logs_zabbixagent.php]: probably should be integrated into MVC, so the log can be downloaded via the API.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            </tuning>
+            <features>
+                <enableActiveChecks type="BooleanField">
+                    <default>1</default>
+                    <Required>Y</Required>
+                </enableActiveChecks>
+                <activeCheckServers type="CSVListField">
+                    <Required>N</Required>
+                    <multiple>Y</multiple>
+                    <mask>/^([a-zA-Z0-9\.:\[\]\s\-]*?,)*([a-zA-Z0-9\.:\[\]\s\-]*)$/</mask>
```
Comment (fabianfrz) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml]: whitespace (\s) in host:port combinations?

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+                </logFileSize>
+                <debugLevel type="OptionField">
+                    <default>val_3</default>
+                    <OptionValues>
+                        <val_0>0 - basic information</val_0>
+                        <val_1>1 - critical information</val_1>
+                        <val_2>2 - error information</val_2>
+                        <val_3>3 - warnings [default]</val_3>
+                        <val_4>4 - debugging</val_4>
+                        <val_5>5 - extended debugging</val_5>
```
Comment (fabianfrz) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml]: probably remove the numbers as the user receives the strings. It is probably better to use round brackets because it would fit better into the rest of the system.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/IndexController.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+class IndexController extends \OPNsense\Base\IndexController
+{
+    /**
+     * zabbix agent index page
+     * @throws \Exception
+     */
+    public function indexAction()
+    {
+        // set page title
+        $this->view->title = "Zabbix Agent Settings";
```
Comment (fabianfrz) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/IndexController.php]: gettext

File: net-mgmt/zabbix-agent/Makefile [hunk 8]
```diff
@@ -0,0 +1,7 @@
+PLUGIN_NAME=		zabbix-agent
+PLUGIN_VERSION=		0.1
+PLUGIN_COMMENT=		Enterprise-class open source distributed monitoring agent
+PLUGIN_DEPENDS=     zabbix32-agent
+PLUGIN_MAINTAINER=	opnsense@moov.de
```
Comment (fabianfrz) [inline net-mgmt/zabbix-agent/Makefile]: maybe align the values

**Comment by fabianfrz** [comment]
please add a package description (same directory as the makefile). This will be shown to the user in the future if he wants more information about the plugin.

File: net-mgmt/zabbix-agent/Makefile [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/Makefile]: PLUGIN_DEVEL=yes

File: net-mgmt/zabbix-agent/Makefile [hunk 10]
```diff
@@ -0,0 +1,7 @@
+PLUGIN_NAME=		zabbix-agent
+PLUGIN_VERSION=		0.1
+PLUGIN_COMMENT=		Enterprise-class open source distributed monitoring agent
+PLUGIN_DEPENDS=     zabbix32-agent
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/Makefile]: is this the actual package that we use? https://github.com/opnsense/tools/blob/master/config/17.1/ports.conf#L88

File: net-mgmt/zabbix-agent/src/etc/inc/plugins.inc.d/zabbixagent.inc [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+    return $services;
+}
+
+/**
+ *  sync configuration via xmlrpc
+ * @return array
+ */
+
+/**
+XXX: needs investigation, hostname must be unique/replaced/excluded on secondary node(s)
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/etc/inc/plugins.inc.d/zabbixagent.inc]: yes, this is really bad for sync. why not use the system hostname instead to avoid the clash?

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/Api/ServiceController.php [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+     * @throws \Exception
+     */
+    public function statusAction()
+    {
+        $backend = new Backend();
+        $mdlAgent = new ZabbixAgent();
+        $response = $backend->configdRun("zabbixagent status");
+
+        if (strpos($response, "not running") > 0) {
+            if ($mdlAgent->settings->main->enabled->__toString() == 1) {
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/Api/ServiceController.php]: although just boilerplate, it's quirky: __toString() == "1"

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/Api/ServiceController.php [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+                $status = "stopped";
+            } else {
+                $status = "disabled";
+            }
+        } elseif (strpos($response, "is running") > 0) {
+            $status = "running";
+        } elseif ($mdlAgent->settings->main->enabled->__toString() == 0) {
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/Api/ServiceController.php]: "0"

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/Api/ServiceController.php [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+            $this->sessionClose();
+
+            $mdlAgent = new ZabbixAgent();
+            $backend = new Backend();
+
+            $runStatus = $this->statusAction();
+
+            // stop zabbix agent when disabled
+            if ($runStatus['status'] == "running" &&
+               ($mdlAgent->settings->main->enabled->__toString() == 0 || $force_restart)) {
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/Api/ServiceController.php]: "0"

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/Api/ServiceController.php [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+                $this->stopAction();
+            }
+
+            // generate template
+            $backend->configdRun('template reload OPNsense/ZabbixAgent');
+
+            // (res)start daemon
+            if ($mdlAgent->settings->main->enabled->__toString() == 1) {
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/Api/ServiceController.php]: ;)

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/IndexController.php (hunk 7 above)
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/IndexController.php]: yessir!

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/forms/settings.xml [hunk 16]
```diff
@@ -0,0 +1,129 @@
+<form>
+    <tab id="zabbixagent-settings-main" description="Main Settings">
+        <field>
+            <id>zabbixagent.settings.main.enabled</id>
+            <label>Enabled</label>
+            <type>checkbox</type>
+            <help><![CDATA[Enable Zabbix Agent service.]]></help>
+        </field>
+        <field>
+            <id>zabbixagent.settings.main.hostname</id>
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/forms/settings.xml]: so yes, why not use the system hostname (preferrably)

wrap the model as "zabbixagent.hostname" and then only sync "zabbix.settings"

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/Menu/Menu.xml [hunk 17]
```diff
@@ -0,0 +1,12 @@
+<menu>
+    <Services>
+      <ZabbixAgent VisibleName="Zabbix Agent" cssClass="fa fa-certificate fa-fw">
+          <Settings order="10" url="/ui/zabbixagent/">
+              <MainSettings VisibleName="Main Settings" url="/ui/zabbixagent/#subtab_zabbixagent-settings-main"/>
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/Menu/Menu.xml]: Can we go 4 levels in the menu? if yes, we shouldn't

File: net-mgmt/zabbix-agent/src/opnsense/scripts/OPNsense/ZabbixAgent/setup.sh [hunk 18]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+AGENT_DIRS="/var/run/zabbix /var/log/zabbix /usr/local/etc/zabbix_agentd.d"
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/scripts/OPNsense/ZabbixAgent/setup.sh]: have you double checked with pkg-plist files? they can have other important dirs.

note this is also mostly for /var if MFS was enabled as /usr/local is persistent and likely created by the package install

https://github.com/opnsense/ports/tree/47aa039c8c8f361193e18a636760392971ef57ae/net-mgmt/zabbix24-server

File: net-mgmt/zabbix-agent/src/opnsense/service/conf/actions.d/actions_zabbixagent.conf [hunk 19]
```diff
@@ -0,0 +1,34 @@
+[setup]
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/service/conf/actions.d/actions_zabbixagent.conf]: is this called separately? if not can zap

File: net-mgmt/zabbix-agent/src/opnsense/service/conf/actions.d/actions_zabbixagent.conf [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+message:stopping zabbix_agentd
+
+[restart]
+command:/usr/local/opnsense/scripts/OPNsense/ZabbixAgent/setup.sh; /usr/local/etc/rc.d/zabbix_agentd restart
+parameters:
+type:script
+message:restarting zabbix_agentd
+
+[status]
+command:/usr/local/etc/rc.d/zabbix_agentd status || exit 0
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/service/conf/actions.d/actions_zabbixagent.conf]: doesn't this return the status if you let it? not as script_output, then check for error

File: net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/rc.conf.d [hunk 21]
```diff
@@ -0,0 +1,6 @@
+{% if helpers.exists('OPNsense.ZabbixAgent.settings.main.enabled') and OPNsense.ZabbixAgent.settings.main.enabled|default("0") == "1" %}
+zabbix_agentd_enable=YES
+{% else %}
+zabbix_agentd_enable=NO
+{% endif %}
+zabbix_agentd_config=/usr/local/etc/zabbix_agentd.conf
```
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/rc.conf.d]: into enable block, setup invoke is also missing:

https://github.com/opnsense/core/blob/master/src/opnsense/service/templates/OPNsense/Proxy/rc.conf.d#L3

File: net-mgmt/zabbix-agent/src/www/diag_logs_zabbixagent.php (hunk 4 above)
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/www/diag_logs_zabbixagent.php]: there is a PR open if anyone wants to touch that.....

File: net-mgmt/zabbix-agent/src/etc/inc/plugins.inc.d/zabbixagent.inc (hunk 1 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/etc/inc/plugins.inc.d/zabbixagent.inc]: yes, it's a warning to myself and others what should be considered before enabling sync :)

File: net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/zabbix_agentd.conf (hunk 3 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/zabbix_agentd.conf]: Removed.

File: net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/zabbix_agentd.conf (hunk 2 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/zabbix_agentd.conf]: I think it is a good habit to keep comments from the sample configuration to give an idea about a certain value and it's usefulness. [It's](https://github.com/opnsense/core/blob/399b9090a40f9ac728863d21b511b75b76759db3/src/etc/inc/system.inc#L1169-L1177) a [very](https://github.com/theforeman/puppet-foreman/blob/ec8c6c25356429b01cd590158932664271db387e/templates/settings.yaml.erb#L30-L31) [common](https://github.com/theforeman/puppet-puppet/blob/ad08281a610d5667c89c2c2d9d95da38c66d08ca/templates/server/puppetserver/conf.d/puppetserver.conf.erb#L24-L33) [thing](https://github.com/jpds/puppet-strongswan/blob/b0a21f9350d41cda437304ab969fe159aba9c868/templates/charon.conf.erb#L6-L14) to do.

File: net-mgmt/zabbix-agent/src/www/diag_logs_zabbixagent.php (hunk 4 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/www/diag_logs_zabbixagent.php]: I think this is outside the scope of this plugin.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml (hunk 6 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml]: Well, in Zabbix the _number_ is actually the debug level. The text is just a vague description. I'm not against removing it, it's just that this is the most precise information...

Regarding the brackets, do you have some examples? I've not seen other GUI elements where the default value was highlighted in any way.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml (hunk 5 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml]: Removed \s.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/IndexController.php (hunk 7 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/IndexController.php]: Fixed. And I will fix this for the "hello world" plugin too ;)

File: net-mgmt/zabbix-agent/Makefile (hunk 8 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/Makefile]: Fixed. Tabs are evil.

File: net-mgmt/zabbix-agent/Makefile (hunk 10 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/Makefile]: There's no point using such an outdated version anymore. I strongly suggest to use 3.2 (or at a minimum 3.0) by now.

File: net-mgmt/zabbix-agent/src/etc/inc/plugins.inc.d/zabbixagent.inc (hunk 11 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/etc/inc/plugins.inc.d/zabbixagent.inc]: We can't just use the hostname. It *must* match the hostname in the Zabbix _Server_, otherwise many checks will not work. That's why it must be configurable.

File: net-mgmt/zabbix-agent/Makefile (hunk 9 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/Makefile]: Added.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml (hunk 6 above)
Comment (fabianfrz) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml]: it is only one entry:
`3 - warnings [default]` -> `3 - warnings (default)`
or even better:
`warnings (default)`

File: net-mgmt/zabbix-agent/Makefile (hunk 10 above)
Comment (fichtner) [inline net-mgmt/zabbix-agent/Makefile]: I misread 32 as 23... ok, will update tools.git

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml (hunk 6 above)
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml]: The most readable is:

"Warnings (3, default)"
"Debugging (4)"
"Extended debugging (5)"

Numbers don't have to be omitted, but we can't assume users know / want to know what numbers correspond to what meaning. It also makes translations clearer.

File: net-mgmt/zabbix-agent/Makefile (hunk 10 above)
Comment (fichtner) [inline net-mgmt/zabbix-agent/Makefile]: https://github.com/opnsense/tools/commit/d2bafa3e0

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml (hunk 6 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml]: Yes, I know what entry you mean... I just wanted to see some examples why round brackets would be better suited. I know at least one example where round brackets would look [really strange](https://github.com/opnsense/plugins/blob/a3520c7c24c4a772201dbbb08fe94f9e31f08084/net/haproxy/src/opnsense/mvc/app/models/OPNsense/HAProxy/HAProxy.xml#L264-L266). And since I couldn't find any other highlighting for default values I've started using this variation for my other plugins already.

File: net-mgmt/zabbix-agent/Makefile (hunk 10 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/Makefile]: Maybe also add a upgrade warning to the release notes, just in case someone is already using Zabbix Agent on OPNsense without a GUI.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/Menu/Menu.xml (hunk 17 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/Menu/Menu.xml]: Huh, what do you mean?

File: net-mgmt/zabbix-agent/Makefile (hunk 10 above)
Comment (fichtner) [inline net-mgmt/zabbix-agent/Makefile]: the package name changes, no problem until they install the plugin, but I'll try to think of something.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/Menu/Menu.xml (hunk 17 above)
Comment (fichtner) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/Menu/Menu.xml]: Level 1: Services
Level 2: Zabbix Agent
Level 3: Settings
Level 4: Main Settings

I don't think we ever had the "Level 4" in the menu?

File: net-mgmt/zabbix-agent/src/opnsense/scripts/OPNsense/ZabbixAgent/setup.sh (hunk 18 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/scripts/OPNsense/ZabbixAgent/setup.sh]: Well, in theory the package creates `%%ETCDIR%%/zabbix_agentd.conf.d`, but it doesn't work for me. I've changed the code to use this (FreeBSD) default directory instead.

File: net-mgmt/zabbix-agent/src/opnsense/service/conf/actions.d/actions_zabbixagent.conf (hunk 19 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/service/conf/actions.d/actions_zabbixagent.conf]: Zapped.

File: net-mgmt/zabbix-agent/src/opnsense/service/conf/actions.d/actions_zabbixagent.conf (hunk 20 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/service/conf/actions.d/actions_zabbixagent.conf]: Fixed. Was using this most likely because of an old bug (in HAProxy plugin).

File: net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/rc.conf.d (hunk 21 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/service/templates/OPNsense/ZabbixAgent/rc.conf.d]: Oh nice! I wasn't aware of this. Added.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/Menu/Menu.xml (hunk 17 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/Menu/Menu.xml]: No no, it's not in the menu, it's just the internal structure. In the GUI it's just `Services -> Zabbix Agent -> Settings` and on the settings page some tabs represent the 4th level.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml (hunk 6 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/models/OPNsense/ZabbixAgent/ZabbixAgent.xml]: @fichtner Nice proposal, I'll change it accordingly.

File: net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/forms/settings.xml (hunk 16 above)
Comment (fraenki) [inline net-mgmt/zabbix-agent/src/opnsense/mvc/app/controllers/OPNsense/ZabbixAgent/forms/settings.xml]: Good idea! Done (with minor changes).

**Comment by fraenki** [comment]
@fabianfrz @fichtner Thanks for your feedback! I've incorporated all changes. Let me know if more should be changed/fixed.

**Comment by fabianfrz** [comment]
looks OK to me.

**Comment by fraenki** [comment]
zabbixagent

**Comment by fichtner** [comment]
@fraenki FYI, I had to do this https://github.com/opnsense/tools/commit/54a0e5616337 due to the bump to 3.x
